### ISSUE TITLE
Global when does not work together with global branches filter [solved]

### ISSUE THREAD

**Issue by lonix1** [woodpecker-ci/woodpecker#1941]
### Component

server

### Describe the bug

The [docs](https://woodpecker-ci.org/docs/next/usage/pipeline-syntax#branch-1) show an example of a global when with `branch` condition:

```yml
when:
  branch: master
``` 

When I tried that, I get an error:
> Execution error
> failed to parse pipeline: could not apply deprecated branches filter into global when filter

### System Info

```shell
version "next-b616a822a0-alpine"
```

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/woodpecker-ci/woodpecker/blob/master/CONTRIBUTING.md).
- [X] Read the [docs](https://woodpecker-ci.org/docs/intro).
- [X] Check that there isn't [already an issue](https://github.com/woodpecker-ci/woodpecker/issues) that reports the same bug to avoid creating a duplicate.
- [X] Checked that the bug isn't fixed in the `next` version already [https://woodpecker-ci.org/faq#which-version-of-woodpecker-should-i-use]
- [X] Check that this is a concrete bug. For Q&A join our [Discord Chat Server](https://discord.gg/fcMQqSMXJy) or the [Matrix room](https://matrix.to/#/#woodpecker:matrix.org).

**Comment by 6543** [comment]
witch forge do you use?

**Comment by 6543** [comment]
this is strange I tested this config and it does exactly what I expect:

```yaml
when:
    branch: main

steps:
    echo:
        image: bash
        commands: echo ja
```

I get a pipeline on `main` but no other branch

**Comment by 6543** [comment]
what event did you use as trigger?

**Comment by 6543** [comment]
tldr: global when was introduced in next (upcomming v1.0.0) and do deprecate the old global branches filter ...

there is a "migration" running to make transition smoth:
https://github.com/woodpecker-ci/woodpecker/blob/45319b24cd4ae148de4cba55725a5a74501ecfb7/pipeline/frontend/yaml/parse.go#L20-L30

but you should not use the old and the new ways at the same time ... dont expect magic :D

**Comment by 6543** [comment]
#1855 will fix that
